modules/public/public.go

```diff
@@ -18,28 +22,39 @@ import (
 
 // Options represents the available options to configure the handler.
 type Options struct {
-	Directory   string
-	Prefix      string
 	CorsHandler func(http.Handler) http.Handler
 }
 
 // AssetsURLPathPrefix is the path prefix for static asset files
 const AssetsURLPathPrefix = "/assets/"
 
+func CustomAssets() *assetfs.Layer {
+	return assetfs.Local("custom", setting.CustomPath, "public")
+}
+
+func StaticAssets() *assetfs.Layer {
+	return assetfs.Local("static", setting.StaticRootPath, "public")
+}
+
+func AssetFS() *assetfs.LayeredFS {
+	if setting.HasBuiltinBindata {
+		return assetfs.Layered(CustomAssets(), StaticAssets(), BuiltinAssets()) // old behavior: always include StaticAssets
```

Comment:
I don't think this is the original logic. The different is the `filesystem` function. In the original code.
For embed mode, the dir will be ignored.
```go
func fileSystem(dir string) http.FileSystem {
	return Assets
}
```
For dynamic mode:
```go
func fileSystem(dir string) http.FileSystem {
	return http.Dir(dir)
}
```

Reply:
Good catch, then the code can be more simple now. Thank you ~~~

bb7b7f7828b5b921539feb5f1af132b075c5261a